File: src/main/java/top/dsbbs2/bukkitcord/api/ICommandRegistry.java
package top.dsbbs2.bukkitcord.api;

import java.util.*;

public interface ICommandRegistry {
    Vector<ICommand> getCommands();
    void setCommands(Vector<ICommand> cmds);
    ICommand setTabCom(ICommand pc, ITabCompleter tc);
    ICommand setComUsa(ICommand pc, String u);
    ICommand setComPer(ICommand pc, String p);
    ICommand setComAlias(ICommand pc, String... alias);
    void regComWithoutRecording(IPlugin plugin, ICommand c);
    void regCom(IPlugin plugin, ICommand c);
    void regComWithCompleter(IPlugin plugin, ICommand c);
    ICommand newPluginCommand(String name, ICommandExecutor ce, IPlugin plugin);
    ICommand newPluginCommand(String name, IPlugin plugin);
}

File: src/main/java/top/dsbbs2/bukkitcord/api/ICommandRegistry.kt
package top.dsbbs2.bukkitcord.api

import java.util.*

interface ICommandRegistry {
    val commands: Vector<ICommand?>?
    fun setTabCom(pc: ICommand?, tc: ITabCompleter?): ICommand?
    fun setComUsa(pc: ICommand?, u: String?): ICommand?
    fun setComPer(pc: ICommand?, p: String?): ICommand?
    fun setComAlias(pc: ICommand?, vararg alias: String?): ICommand?
    fun regComWithoutRecording(plugin: IPlugin?, c: ICommand?)
    fun regCom(plugin: IPlugin?, c: ICommand?)
    fun regComWithCompleter(plugin: IPlugin?, c: ICommand?)
    fun newPluginCommand(name: String?, ce: ICommandExecutor?, plugin: IPlugin?): ICommand?
    fun newPluginCommand(name: String?, plugin: IPlugin?): ICommand?
}